### PR TITLE
Fix creating subnet with custom ID for VPC with custom ID throwing 500 internal server error

### PR DESCRIPTION
**Pull request by RobertLucian**
#### Description

When running:

```bash
awslocal ec2 create-vpc --cidr-block 10.0.0.0/16 --tag-specifications "ResourceType=vpc,Tags=[{Key=_custom_id_,Value=vpc-lsdefault}]"
awslocal ec2 describe-vpcs
awslocal ec2 create-subnet --vpc-id vpc-lsdefault --cidr-block 10.0.0.0/16 --tag-specifications "ResourceType=subnet,Tags=[{Key=_custom_id_,Value=mysubnet}]" 
```

The following error would be returned:

```python
2024-09-05T15:06:51.529  INFO --- [et.reactor-0] localstack.request.aws     : AWS ec2.CreateSubnet => 500 (InternalError)
2024-09-05T15:06:52.721 ERROR --- [et.reactor-0] l.aws.handlers.logging     : exception during call chain
Traceback (most recent call last):
  File "/opt/code/localstack/.venv/lib/python3.11/site-packages/rolo/gateway/chain.py", line 166, in handle
    handler(self, self.context, response)
  File "/opt/code/localstack/.venv/lib/python3.11/site-packages/localstack/aws/handlers/service.py", line 113, in __call__
    handler(chain, context, response)
  File "/opt/code/localstack/.venv/lib/python3.11/site-packages/localstack/aws/handlers/service.py", line 83, in __call__
    skeleton_response = self.skeleton.invoke(context)
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/code/localstack/.venv/lib/python3.11/site-packages/localstack/aws/skeleton.py", line 154, in invoke
    return self.dispatch_request(serializer, context, instance)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/code/localstack/.venv/lib/python3.11/site-packages/localstack/aws/skeleton.py", line 168, in dispatch_request
    result = handler(context, instance) or {}
             ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/code/localstack/.venv/lib/python3.11/site-packages/localstack/aws/forwarder.py", line 133, in _call
    return handler(context, req)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/opt/code/localstack/.venv/lib/python3.11/site-packages/localstack/aws/skeleton.py", line 118, in __call__
    return self.fn(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/code/localstack/.venv/lib/python3.11/site-packages/localstack/aws/api/core.py", line 160, in operation_marker
    return fn(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^
  File "/opt/code/localstack/.venv/lib/python3.11/site-packages/localstack/services/ec2/provider.py", line 286, in create_subnet
    response = call_moto(context)
               ^^^^^^^^^^^^^^^^^^
  File "/opt/code/localstack/.venv/lib/python3.11/site-packages/localstack/services/moto.py", line 48, in call_moto
    return dispatch_to_backend(context, dispatch_to_moto, include_response_metadata)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/code/localstack/.venv/lib/python3.11/site-packages/localstack/aws/forwarder.py", line 185, in dispatch_to_backend
    http_response = http_request_dispatcher(context)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/code/localstack/.venv/lib/python3.11/site-packages/localstack/services/moto.py", line 118, in dispatch_to_moto
    response = dispatch(request, raw_url, request.headers)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/code/localstack/.venv/lib/python3.11/site-packages/moto/core/responses.py", line 290, in dispatch
    return cls()._dispatch(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/code/localstack/.venv/lib/python3.11/site-packages/moto/core/responses.py", line 502, in _dispatch
    return self.call_action()
           ^^^^^^^^^^^^^^^^^^
  File "/opt/code/localstack/.venv/lib/python3.11/site-packages/moto/core/responses.py", line 588, in call_action
    response = method()
               ^^^^^^^^
  File "/opt/code/localstack/.venv/lib/python3.11/site-packages/moto/ec2/responses/subnets.py", line 20, in create_subnet
    subnet = self.ec2_backend.create_subnet(
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/code/localstack/.venv/lib/python3.11/site-packages/localstack/utils/patch.py", line 61, in proxy
    return new(target, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/code/localstack/.venv/lib/python3.11/site-packages/localstack/services/ec2/patches.py", line 38, in ec2_create_subnet
    result: ec2_models.subnets.Subnet = fn(self, *args, tags=tags, **kwargs)
                                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/code/localstack/.venv/lib/python3.11/site-packages/moto/ec2/models/subnets.py", line 409, in create_subnet
    self.associate_default_network_acl_with_subnet(subnet_id, vpc_id)  # type: ignore[attr-defined]
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/code/localstack/.venv/lib/python3.11/site-packages/moto/ec2/models/network_acls.py", line 184, in associate_default_network_acl_with_subnet
    acl = next(
          ^^^^^
StopIteration
```

This PR fixes this specific issue.

### CODE REVIEW

**Review by viren-nadkarni — COMMENTED** [review]
`VPCBackend.create_vpc(...)` (which we patch but also [invoke the original](https://github.com/localstack/localstack/blob/759cce52e335e2d92ded421558058492624a5db6/localstack-core/localstack/services/ec2/patches.py#L100)) [calls](https://github.com/localstack/moto/blob/ec314fbfcebef72ad6b526da11477e1a202dcbd7/moto/ec2/models/vpcs.py#L749) `create_network_acl(...)`.

Have you looked into why that is not working? I don't think we should be creating ACLs from subnet operations because it falls in VPC's realm.

File: localstack-core/localstack/services/ec2/patches.py [hunk 1]
```diff
@@ -109,11 +110,29 @@ def ec2_create_vpc(
...
 
+            # Create default network ACL, route table, and security group for custom ID VPC
+            self.create_route_table(
+                vpc_id=custom_id,
+                main=True,
+            )
+            self.create_network_acl(
+                vpc_id=custom_id,
+                default=True,
+            )
```
Comment (dominikschubert) [inline localstack-core/localstack/services/ec2/patches.py:135]: Nice, definitely in a better position now.

File: localstack-core/localstack/services/ec2/patches.py [hunk 2]
```diff
@@ -109,11 +110,29 @@ def ec2_create_vpc(
                     vpc_id=vpc_id,
                 )
 
+            # Delete route table if only main route table remains.
+            route_tables = self.describe_route_tables(filters={"vpc-id": vpc_id})  # type: ignore[attr-defined]
+            if len(route_tables) > 1:
+                raise DependencyViolationError(
+                    f"The vpc {vpc_id} has dependencies and cannot be deleted."
```
Comment (dominikschubert) [inline localstack-core/localstack/services/ec2/patches.py:117]: Why the VPC? Shouldn't this be the route table?

**Review by dominikschubert — APPROVED** [review]
If tests pass, let's move head with merging and maybe @viren-nadkarni  can give it a quick glance tomorrow as well to see if it's aligned with his vision for the service :+1:

File: localstack-core/localstack/services/ec2/patches.py (hunk 2 above)
Comment (RobertLucian) [reply localstack-core/localstack/services/ec2/patches.py]: Good catch. Simplified the deletion of the route tables.

**Pull request merged**